src/base64.cr

```diff
@@ -156,36 +156,196 @@ module Base64
     strict_encode_to_io_internal(data, io, CHARS_SAFE, pad: true)
   end
 
-  # Returns the base64-decoded version of *data* as a `Bytes`.
+  # Writes the base64-decoded version of `from` into `to`,
+  # and returns the amount of bytes written.
+  #
+  # If the `to` buffer is not big enough to store the resulting bytes,
+  # or if the input is invalid base64, the method raises a `Base64::Error`.
+  #
+  # This will decode either the normal or urlsafe alphabets.
+  def decode(from : Bytes, to : Bytes) : Int32
+    total_written_bytes : Int32 = 0
+    total_read_bytes : Int32 = 0
+
+    from_ptr = from.to_unsafe
+    from_size = from.size
+    to_ptr = to.to_unsafe
+    to_size = to.size
```

Comment:
I doubt it makes much sense to put these values into local variables. The `Bytes` parameters are passed as structs on the stack and thus are already very local.